src/prefect/workers/base.py

```diff
@@ -1252,7 +1253,12 @@ async def _mark_flow_run_as_cancelled(
         state_updates.setdefault("type", StateType.CANCELLED)
         if TYPE_CHECKING:
             assert flow_run.state
-        state = flow_run.state.model_copy(update=state_updates)
+
+        if flow_run.state:
+            state = flow_run.state.model_copy(update=state_updates)
+        else:
+            # Unexpectedly when flow run does not have a state, create a new one
+            state: State = State(**state_updates)  # pyright: ignore[reportAssignmentType]
```

Comment:
There's a utility in `prefect.states` that you can use to easily create a `CANCELLED` state and appease `pyright`.

```suggestion
            state: State = Cancelled()
```

Reply:
Updated [`dc8c0da` (#17537)](https://github.com/PrefectHQ/prefect/pull/17537/commits/dc8c0dae081032050f9b40bce39959e9efb167c6)